src/Form/Traits/LinkFieldGetOwnerTrait.php

```diff
@@ -18,4 +18,19 @@ private function getOwner(): DataObject
         }
         return $owner;
     }
+
+    private function getOwnerFields(): array
+    {
+        $owner = $this->getOwner();
+        $relation = $this->getName();
+        // Elemental content block
+        if (preg_match('#^PageElements_[0-9]+_(.+)$#', $relation, $matches)) {
+            $relation = $matches[1];
+        }
```

Comment:
Use `ElementalAreaController::removeNamespacesFromFields()` instead.
```suggestion
        if ($owner instanceof  BaseElement) {
            $someVariableName = ElementalAreaController::removeNamespacesFromFields([$relation => ''], $owner->ID);
            $relation = array_keys($someVariableName)[0];
        }
```
I haven't tested that code but something like that should work.

Reply:
We probably also need a unit test for this

Reply:
Done